### DISCUSSION TITLE
How to get sso user token in messaging extension?

### DISCUSSION THREAD

**Discussion by Swtplanet** [microsoft/teams-sdk#460] [Q&A]
Please help on how can we get sso based usertoken in messaging extension.
I have done setup mentioned on below url.
https://learn.microsoft.com/en-us/microsoftteams/platform/sbs-bots-with-sso?tutorial-step=3

I have added below in app.

authentication: {
        connectionName: "Connection Name given in Auth setting in bot configuration",
        title: "Sign In"
    }

I have made connection using client id of bot in Configuration of bot on azure.

I am trying to get authtoken using state.temp.value.authtoken . After doing above steps, I get authtoken as undefined.

Please give example, on how should we implement this. Also, please tell whether I am supposed to get any sign-in dialog?

**Comment by corinagum** [comment]
@singhk97

**Comment by blackchoey** [comment]
I created an example to use `handleMessageExtensionQueryWithSSO` from `@microsoft/TeamsFx` package to implement SSO: https://github.com/blackchoey/message-extension-sso. Currently the project can only run locally (by pressing F5 with Teams Toolkit extension installed in VS Code). You can have a try. [this commit](https://github.com/blackchoey/message-extension-sso/commit/6523c752097ab4f3703a6e600129462aaeb645b7) shows the tasks to be done to enable SSO.

This is not the final solution in Teams-AI library. Please stay tuned for the updates of the library.

**Reply by Swtplanet** [reply]
Thank you for example.
Just wanted to understand , can I get token using this method in Message Extension’s Fetch and SubmitAction. I am trying action type of messaging extension.

**Reply by blackchoey** [reply]
No. Currently the `handleMessageExtensionQueryWithSSO` function only works for query and there's no other similar function for action commands temporary.

**Reply by Swtplanet** [reply]
Ok, currently, in this case, we cannot get sso token for action type of messaging extension. We need to wait for updates in library.
Can you confirm ?
